plugins/git/git.plugin.zsh

```diff
@@ -32,6 +32,22 @@ function work_in_progress() {
     echo "WIP!!"
   fi
 }
+#change current branch old email with new eamil
+gmvemail() {
+  export GMVEMAIL_GIT_EMAIL_OLD=$1
+  export GMVEMAIL_GIT_EMAIL_NEW=$2
```

Comment:
Ok, then set up that inside a subshell:
```zsh
(
  export GMVEMAIL_GIT_EMAIL_OLD=$1
  export GMVEMAIL_GIT_EMAIL_NEW=$2
  git filter-branch ...
)
```

or just set the variables before the command, like so:
```zsh
OLD_EMAIL=$1 NEW_EMAIL=$2 git filter-branch --commit-filter '
    ...
' HEAD
```

I like the second one better.